Objective-C/CBLArray.mm

```diff
@@ -59,21 +66,33 @@ - (instancetype) initWithCopyOfMArray: (const MArray<id>&)mArray
     self = [super init];
     if (self) {
         _array.initAsCopyOf(mArray, isMutable);
+        _sharedLock = [self _sharedLock];
     }
     return self;
 }
 
 
+- (NSObject*) _sharedLock {
```

Comment:
Since this is only called from the init methods, it seems more like an init method itself. Why not make it `-init`, and call it as `[self init]` at the start of the other initializers?